be/src/storage/lake/tablet_manager.cpp

```diff
@@ -267,6 +271,75 @@ Status TabletManager::put_tablet_metadata(const TabletMetadata& metadata) {
     return put_tablet_metadata(std::move(metadata_ptr));
 }
 
+// NOTE: tablet_metas is non-const and we will clear schemas for optimization.
+// Callers should ensure thread safety.
+Status TabletManager::put_aggregate_tablet_metadata(std::map<int64_t, TabletMetadataPB>& tablet_metas) {
+    if (tablet_metas.empty()) {
+        return Status::InternalError("tablet_metas cannot be empty");
+    }
+
+    SharedTabletMetadataPB shared_meta;
+    auto& first_meta = tablet_metas.begin()->second;
+    const int64_t schema_id = first_meta.schema().id();
+    shared_meta.set_schema_id(schema_id);
+
+    std::unordered_map<int64_t, TabletSchemaPB> unique_schemas;
+    for (auto& [tablet_id, meta] : tablet_metas) {
+        if (schema_id != meta.schema().id()) {
+            return Status::InternalError("tablet schema not the same");
+        }
+        unique_schemas.emplace(meta.schema().id(), meta.schema());
+    }
+
+    for (auto& [tablet_id, meta] : tablet_metas) {
+        for (const auto& [ver, schema] : meta.historical_schemas()) {
+            unique_schemas.emplace(ver, schema);
+        }
+    }
+
+    for (auto& [schema_id, schema] : unique_schemas) {
+        (*shared_meta.mutable_schemas())[schema_id] = std::move(schema);
+    }
+
+    auto make_page_pointer = [](int64_t offset, int64_t size) {
+        PagePointerPB pointer;
+        pointer.set_offset(offset);
+        pointer.set_size(size);
+        return pointer;
+    };
+
+    const std::string meta_location =
+            aggregate_tablet_metadata_location(tablet_metas.begin()->first, first_meta.version());
+    ProtobufFile meta_file(meta_location);
```

Comment:
Why we need to introduce `ProtobufFile` here? if we use normal `WritableFile`, it can still cover this case, right?